hangman.py

```diff
@@ -0,0 +1,340 @@
+                      get_guessed_word(secret_word, letters_guessed))
+                continue
+            else:
+                warnings_left -= 1
+                print("Oops! You've already guessed that letter. You have", warnings_left, "warnings left:",
+                      get_guessed_word(secret_word, letters_guessed))
+                continue
+        else:
+            letters_guessed.append(guessed_letter)
+
+        if guessed_letter in letters_in_secret_word:
+            #Part of code for correct guess
+            print("Good guess:", get_guessed_word(secret_word, letters_guessed))
+        else:
+            #System of losing guesses for incorrect guessed word
+            if guessed_letter in list_of_en_vowels:
+                guesses_left -= 2
+            else:
+                guesses_left -= 1
+            print("Oops! That letter is not in my word:", get_guessed_word(secret_word, letters_guessed))
+
+        if is_word_guessed(secret_word, letters_guessed):
+            #End of the game (winning), winning = True
+            winning = True
+            break
+
+    #Showing information about game results
+    if winning:
+        total_score = guesses_left * len_secret_word_without_dublicate(secret_word)
+        print("Congratulations, you won! Your total score for this game is:", total_score)
+    else:
+        print("Sorry, you ran out of guesses. The word was", secret_word)
+
+
+
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the 
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise: 
+    '''
+
+    #Two lists above are needed to delete  space gaps in my_word
+    #Then they are used to compare lengths of my_word without space gaps and then compare every symbols
+
+
+    my_word = my_word.replace('_ ', '_')
+
+    if len(my_word) != len(other_word):
+        return False
+
+    for symbol in range(0, len(my_word)):
+        if my_word[symbol] == '_' or my_word[symbol] == other_word[symbol]:
+            continue
+        return False
+
+   #Code below is needed for not showing words, if they have same letters and length, but my_word has any letter and other_word has two same letters as letter in my_word
+   #For example, function doesn't show word 'apple' if my_word is 'a_ ple'
+
+    my_word = my_word.replace('_', '')
+
+    for symbol in range(0, len(my_word)):
+        #Counting number of every letter in both words and comparing them
+        letters_1 = my_word.count(my_word[symbol])
+        letters_2 = other_word.count(my_word[symbol])
+        if letters_1 == letters_2:
+            continue
+        else:
+            return False
```

Comment:
Effectively - you could invert `if` statement and forget about using `else` here

Reply:
done